database/location.go

```diff
@@ -0,0 +1,17 @@
+package database
+
+// StoreLocation represents a location of a data chunk located at a database store
+type StoreLocation []byte
+
+// Serialize returns a byte slice that represents the StoreLocation
+func (s StoreLocation) Serialize() []byte {
+	serializedLocation := make([]byte, len(s))
+	copy(serializedLocation, s)
+	return serializedLocation
+}
+
+// Deserialize deserializes the given byte slice into s.
+func (s *StoreLocation) Deserialize(serializedLocation []byte) {
```

Comment:
Why not make Deserialize a package-level function that returns StoreLocation? (ofc, rename it to DeserializeStoreLocation or something similar)

Reply:
Also: I think we want to check that serializedLocation is a valid representation of StoreLocation